hibernate-core/src/main/java/org/hibernate/tool/schema/internal/ColumnDefinitions.java

```diff
@@ -249,6 +256,25 @@ private static String normalize(String typeName) {
 		}
 	}
 
+	private static String extractTypeName(String typeExpression) {
+		String sqlType = stripConstraints(typeExpression);
+		String typeName = stripArgs( sqlType );
+		typeName = normalize( typeName );
+		return typeName;
+	}
+
+	static String stripConstraints(String typeExpression) {
+		String sqlType = typeExpression.toLowerCase(Locale.ROOT);
+		sqlType = keepBefore( sqlType, " default " ); // Strip default value
+		sqlType = keepBefore( sqlType, " not null" );
```

Comment:
I mean, there really are a lot more possibilities than just `default` and `not null`. What about `generated`? What about `unique`? What about `constraint`? `primary key`? etc, etc.

Reply:
I 100% agree that it's bad. `columnDefinition` was a TopLink thing that I never really liked but at the time of JPA 1.0 didn't have a good reason to object to.

As I've said, we need to support it because it's in the spec, but there's no requirement that migration/validation handle it elegantly.